src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3438 @@
+
+    /* 25 */ ENAN_ANIMOBJ_AN4,
+    /* 25 */ ENAN_ANIM_MASK_STAND_LOOP = ENAN_ANIMOBJ_AN4,
+    /* 26 */ ENAN_ANIM_MASK_KNEEL,
+    /* 27 */ ENAN_ANIM_MASK_KNEEL_LOOP,
+    /* 28 */ ENAN_ANIM_HUG,
+    /* 29 */ ENAN_ANIM_HUG_LOOP,
+    /* 30 */ ENAN_ANIM_HUG_RELEASE,
+    /* 31 */ ENAN_ANIM_HUG_RELEASE_LOOP,
+    /* 32 */ ENAN_ANIM_COMBINE_MASKS_1,
+    /* 33 */ ENAN_ANIM_COMBINE_MASKS_2,
+    /* 34 */ ENAN_ANIM_LOOK_UP,
+    /* 35 */ ENAN_ANIM_LOOK_UP_LOOP,
+    /* 36 */ ENAN_ANIM_MAX
+} EnAnAnimation;
+
+static AnimationInfoS sAnimationInfo[ENAN_ANIM_MAX] = {
+    { &gAnju1HoldHandsAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },          // ENAN_ANIM_HOLD_HANDS
+    { &gAnju1IdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },               // ENAN_ANIM_IDLE
+    { &gAnju1IdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -6 },              // ENAN_ANIM_IDLE_MORPH
+    { &gAnju1BowAnim, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },                // ENAN_ANIM_BOW
+    { &gAnju1BowAnim, 1.0f, 0, -1, ANIMMODE_ONCE, -6 },               // ENAN_ANIM_BOW_MORPH
+    { &gAnju1SurprisedAnim, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },          // ENAN_ANIM_SURPRISED
+    { &gAnju1LookUpRelievedAnim, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },     // ENAN_ANIM_LOOK_UP_RELIEVED
+    { &gAnju1WalkAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },               // ENAN_ANIM_WALK
+    { &gAnju1WalkAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },               // ENAN_ANIM_WALK2
+    { &gAnju1SitAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },                // ENAN_ANIM_SIT
+    { &gAnju1SitAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -6 },               // ENAN_ANIM_SIT_MORPH
+    { &gAnju1SittingInDisbeliefAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 }, // ENAN_ANIM_SITTING_IN_DISBELIEF
+    { &gAnju1CookAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },               // ENAN_ANIM_COOK
+    { &gAnju1TasteAnim, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },              // ENAN_ANIM_TASTE
+    { &gAnju1TasteEndAnim, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },           // ENAN_ANIM_TASTE_END
+    { &gAnju1WaitWithTrayAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },       // ENAN_ANIM_WAIT_WITH_TRAY
+    { &gAnju1WalkWithTrayAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },       // ENAN_ANIM_WALK_WITH_TRAY
+
+    // ENAN_ANIMOBJ_AN2
+    { &gAnju2UmbrellaIdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 }, // ENAN_ANIM_UMBRELLA_IDLE
+    { &gAnju2UmbrellaWalkAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 }, // ENAN_ANIM_UMBRELLA_WALK
+    { &gAnju2UmbrellaSitAnim, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },  // ENAN_ANIM_UMBRELLA_SIT
+    { &gAnju2UmbrellaCryAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },  // ENAN_ANIM_UMBRELLA_CRY
+
+    // ENAN_ANIMOBJ_AN3
+    { &gAnju3BroomIdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },  // ENAN_ANIM_BROOM_IDLE
+    { &gAnju3BroomWalkAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },  // ENAN_ANIM_BROOM_WALK
+    { &gAnju3BroomSweepAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 }, // ENAN_ANIM_BROOM_SWEEP
+    //! @bug Uses symbol from OBJECT_AN2 instead of OBJECT_AN3
+    { &gAnju2UmbrellaSitAnim, -1.0f, 0, -1, ANIMMODE_ONCE, 0 }, // ENAN_ANIM_24
```

Comment:
Maybe it would be good to explain what happens if `ENAN_ANIM_24` is used? I think the game will just crash, but I don't think this comment gets that across.

Reply:
The game just crashes yeah.

Here I changed when Anju uses the BROOM_IDLE animation to use this bugged one:

https://github.com/zeldaret/mm/assets/7416381/139678be-e889-4b3e-93c8-e1a9049baf06





```diff
diff --git a/src/overlays/actors/ovl_En_An/z_en_an.c b/src/overlays/actors/ovl_En_An/z_en_an.c
index 08d73fa4c..e17ceeec9 100644
--- a/src/overlays/actors/ovl_En_An/z_en_an.c
+++ b/src/overlays/actors/ovl_En_An/z_en_an.c
@@ -1987,7 +1987,7 @@ s32 EnAn_HandleDialogue(EnAn* this, PlayState* play) {
 
                 case 0x28F7: // Everyone will be taking refuge by tomorrow
                 case 0x28F8: // We'll go to Romani Ranch to take refuge
-                    EnAn_ChangeAnim(this, play, ENAN_ANIM_BROOM_IDLE);
+                    EnAn_ChangeAnim(this, play, ENAN_ANIM_24);
                     break;
 
                 case 0x28EB: // I'm afraid to meet him...
@@ -2099,7 +2099,7 @@ s32 EnAn_HandleDialogue(EnAn* this, PlayState* play) {
                         EnAn_ChangeAnim(this, play, ENAN_ANIM_UMBRELLA_IDLE);
                     }
                     if ((this->animIndex == ENAN_ANIM_BROOM_WALK) || (this->animIndex == ENAN_ANIM_BROOM_SWEEP)) {
-                        EnAn_ChangeAnim(this, play, ENAN_ANIM_BROOM_IDLE);
+                        EnAn_ChangeAnim(this, play, ENAN_ANIM_24);
                     }
                     if ((this->animIndex == ENAN_ANIM_WALK) || (this->animIndex == ENAN_ANIM_WALK2)) {
                         EnAn_ChangeAnim(this, play, ENAN_ANIM_IDLE_MORPH);
@@ -3332,6 +3332,8 @@ void EnAn_Init(Actor* thisx, PlayState* play) {
     }
 
     this->actionFunc = EnAn_Initialize;
+
+    gSaveContext.save.day = 3;
 }
 
 void EnAn_Destroy(Actor* thisx, PlayState* play) {
```

Reply:
What it actually happens:
- Requesting this enum will try to use the symbol `0x060038A0`
- This enum is in the `object_an3` block of enum values, so this segmented symbol will reference said object
- but `object_an3` is only `0x2A60` bytes long, so this is an OoB read, yeeey

Because this is an OoB read, I guess this isn't a guaranteed crash all the time, and console may behave differently compared to pj64 (i should move to ares, but i'm too lazy to set it up), but either way a bad thing will happen